guillotina/db/transaction.py

```diff
@@ -407,8 +452,8 @@ def tpc_cleanup(self):
     async def items(self, container):
         # XXX not using cursor because we can't cache with cursor results...
         keys = await self.keys(container._p_oid)
```

Comment:
Would not be better to have an async generator for the keys and get the batches with the keys?

Reply:
async generators in postgresql queries are not ideal. They require cursors. Cursors require an active transaction and we often don't have that.